SourceCodeTests/ExecutableFileTests.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Shouldly;
+using System;
+using System.IO;
+using Xunit;
+
+namespace FileTests
+{
+    public class ExecutableFileTests
+    {
+        public byte[] FileContent { get; set; }
+        public bool IsExecutable { get; set; } = true;
+
+        public ExecutableFileTests()
+        {
+            var filePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "..\\..\\..\\..\\SourceCodeTests\\FileStorage\\SourceCodeFile.txt");
+            FileContent = File.ReadAllBytes(filePath);
+        }
+
+        [Fact]
+        public void Check_If_Executable_File()
+        {
+            byte[] result = new byte[2];
+            Array.Copy(FileContent, 0, result, 0, 2);
+
+            try
```

Comment:
Test logic should have 0 nesting - this means no branches, loops, or try-catch.

Shouldly supports testing if a function throws an exception - https://docs.shouldly.io/documentation/exceptions/throw#79612a9b